repos/system_upgrade/common/actors/peseventsscanner/libraries/pes_events_scanner.py

```diff
@@ -480,6 +480,21 @@ def apply_transaction_configuration(source_pkgs):
     return source_pkgs_with_conf_applied
 
 
+def remove_leapp_related_events(events):
+    leapp_pkgs = [
+        'leapp', 'leapp-deps', 'leapp-upgrade-el7toel8', 'leapp-upgrade-el8toel9',
+        'leapp-upgrade-el7toel8-deps', 'leapp-upgrade-el8toel9-deps', 'python2-leapp',
+        'python3-leapp', 'snactor'
+    ]
```

Comment:
https://github.com/oamg/leapp-repository/blob/d9af1f2a19ec3352a4eff596bcb13e7ad073d763/repos/system_upgrade/common/libraries/dnfconfig.py#L6 - we have this function in the shared libraries. I think it's better to use this function to get the list of packages so it's not defined on multiple places.  However, the set of packages is superset of the one got from the function, and the whole set is not expected to be obtained from that one function. So I am thinking about to introduce a new one, like `get_all_leapp_packages()` or a parameter to be able to obtain the whole set. The point is that we have already 3 situations when we need a list of our packages (usually not the full list). And seems to me that it would be great to have just a one place where it's defined. @fernflower  is dealing with it now also in PR #1148 .

@oamg/developers  any ideas?

Reply:
Having a single place where leapp package names are defined is a way to go. Implementation-wise I would go with
```python3
def get_leapp_packages(kind: LeappPackageKind):
    if kind == LeappPackageKind.EVERY_IN_UNIVERSE:
        pass
    elif kind == LeappPackageKind.ALL_FOR_THIS_UPG:
        pass
```
where `LeappPackageKind` is an enum. This way it forces the caller to think about what leapp packages are needed, while the approach is extensible - the enum can always be extended with new "classes" of packages that are discovered to be needed in a situation. Having 2 separate functions is, imho, error-prone.

Reply:
We've already had an internal discussion about this on Cabal, there will be a follow up email.

Reply:
@MichalHe  in case you would like to take a look before I send the email (in the morning), check the cabal doc - mtg minutes. We operates now with something like `get_leapp_packages(..., <kind>=[...]`. We are discussing kind of the taxonomy, but enum class is the final way I would expect to go. just we do not know what will be particular entries - there will be voting on that on Thu in the cabal doc. proposals can be added during whole Wed :)

Reply:
Love the enum idea, I'd only use `package_type` instead of `kind`.